prov/sockets/src/sock_av.c

```diff
@@ -547,7 +564,12 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 			ret = -FI_EINVAL;
 			goto err2;
 		}
+		_av->table = (struct sock_av_addr *)((char *)_av->table_hdr +
+				idx_arr_sz +
+				sizeof(struct sock_av_table_hdr));
 	} else {
+		table_sz = sizeof(struct sock_av_table_hdr) +
+			_av->attr.count * sizeof(struct sock_av_addr);
```

Comment:
There are several variants and duplicates of the same calculation.  Please look for a way to clean these up and isolate to common code, possibly introducing some macro's.